pkg/pool-weighted/contracts/LBPool.sol

```diff
@@ -0,0 +1,312 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import { SafeCast } from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+
+import { IBasePoolFactory } from "@balancer-labs/v3-interfaces/contracts/vault/IBasePoolFactory.sol";
+import { IRouterCommon } from "@balancer-labs/v3-interfaces/contracts/vault/IRouterCommon.sol";
+import { IVaultErrors } from "@balancer-labs/v3-interfaces/contracts/vault/IVaultErrors.sol";
+import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
+import "@balancer-labs/v3-interfaces/contracts/vault/VaultTypes.sol";
+
+import { InputHelpers } from "@balancer-labs/v3-solidity-utils/contracts/helpers/InputHelpers.sol";
+import { FixedPoint } from "@balancer-labs/v3-solidity-utils/contracts/math/FixedPoint.sol";
+import { BaseHooks } from "@balancer-labs/v3-vault/contracts/BaseHooks.sol";
+
+import { GradualValueChange } from "./lib/GradualValueChange.sol";
+import { WeightedPool } from "./WeightedPool.sol";
+
+/**
+ * @notice Weighted Pool with mutable weights, designed to support v3 Liquidity Bootstrapping.
+ * @dev Inheriting from WeightedPool is only slightly wasteful (setting 2 immutable weights and `_totalTokens`,
+ * which will not be used later), and it is tremendously helpful for pool validation and any potential future
+ * base contract changes.
+ */
+contract LBPool is WeightedPool, Ownable, BaseHooks {
+    using SafeCast for *;
+
+    // Since we have max 2 tokens and the weights must sum to 1, we only need to store one weight.
+    // Weights are 18 decimal floating point values, which fit in less than 64 bits. Store smaller numeric values
+    // to ensure the PoolState fits in a single slot.
+    struct PoolState {
+        uint32 startTime;
+        uint32 endTime;
+        uint64 startWeight0;
+        uint64 endWeight0;
+        bool swapEnabled;
+    }
+    // `{start,end}Time` are `uint32`s. Ensure that no input time (passed as `uint256`) will overflow.
+    uint256 private constant _MAX_TIMESTAMP = type(uint32).max;
```

Comment:
```suggestion
```

If we're using SafeCast now, it should be safe to axe this line and associated comment

Reply:
Yes; I guess we're not using this anymore. I'd originally kept it as documentation, but I can just expand the comment to show this. Good catch! (I think I did a search and thought it was being used - but it was the version in FactoryWidePauseWindow, which is private and can't be referenced from here, and I certainly don't want to change anything there.)